### ISSUE TITLE
Need a logging level between DEBUG and INFO

### ISSUE THREAD

**Issue by jahagirdar** [cocotb/cocotb#1226]
Logging level DEBUG prints lots of information about cocotb internals which is not useful for a testcase writer or a dut debugger.
Usage of Levels above INFO make sense only if we want to alert the user about some potential issue (unimplemented features, test errors)
So for a testcase writer INFO gets crammed with two types of information.

1. General information( e.g. Test state like "Reset Done" "Sending Packet No. X", "Received Packet Y" etc.)
2. Test Debug Information (Packet content was {list of bytes}, Configured register X with value Y etc)
Creating a loglevel between DEBUG and INFO will help in separating these two usecases.

**Comment by eric-wieser** [comment]
You can set the level per logger in your test module, outside of any tests. Does that help?

**Comment by eric-wieser** [comment]
Perhaps we need some better docs here

**Comment by cmarqu** [comment]
Related: #493

**Comment by jahagirdar** [comment]
@eric-wieser 

> You can set the level per logger in your test module, outside of any tests. Does that help?

In addition to setting the log level we may require a separate `cocotb.log.debug` and `cocotb.log.core_debug` function

**Comment by eric-wieser** [comment]
For logging your own things, i'd recommend:
```
my_logger = logging.getLogger(__name__)
my_logger.setLevel(logging.DEBUG)

# Use my_logging.debug below
```

**Comment by cmarqu** [comment]
Some or all of this seems to be solved by #2502 now.
